### PR TITLE
fix/iplog_history_rotate_index

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Fixes issue where iplog rotate is taking a log time
## Impacts

Database indexes
## Issue

Issue #1754
## NEWS file entries

Enhancements
++++++++++++
- Improve speed of iplog_history rotation
## Delete branch after merge

NO

### CODE REVIEW

**Comment by dwlfrth** [comment]
Let's add it to the 'iplog_archive' table as well.
Can we also add one on the 'start_time' ?

**Comment by jrouzierinverse** [comment]
Are we performing any searches on the start_time for iplog_history and iplog_archive?

**Comment by dwlfrth** [comment]
Yes

**Comment by extrafu** [comment]
Rebase

**Review by dwlfrth — CHANGES_REQUESTED** [review]
Please add index on "start_time" also.
Both indexes should also exists in "iplog_archive"

**Comment by jrouzierinverse** [comment]
done

File: db/upgrade-X.X.X-X.Y.Z.sql [hunk 1]
```diff
@@ -316,3 +316,15 @@ DROP INDEX mac ON violation;
 --
 
 DROP INDEX ip on inline_accounting;
+
+--
+-- Alter table iplog_history
+--
+
+ALTER TABLE iplog_history ADD INDEX end_time (end_time) ADD INDEX start_time (end_time);
```
Comment (dwlfrth) [inline db/upgrade-X.X.X-X.Y.Z.sql]: second column name need to be adjusted

File: db/upgrade-X.X.X-X.Y.Z.sql [hunk 2]
```diff
@@ -316,3 +316,15 @@ DROP INDEX mac ON violation;
...
+
+--
+-- Alter table iplog_archive
+--
+
+ALTER TABLE iplog_archive ADD INDEX end_time (end_time) ADD INDEX start_time (end_time);
```
Comment (dwlfrth) [inline db/upgrade-X.X.X-X.Y.Z.sql]: second column name need to be adjusted

**Review by dwlfrth — CHANGES_REQUESTED** [review]

**Comment by jrouzierinverse** [comment]
fixed

James Rouzier
jrouzier@inverse.ca :: +1.514.447.4918 (x115)  ::  http://www.inverse.ca
Inverse inc. :: Leaders behind SOGo (http://www.sogo.nu) and PacketFence (http://www.packetfence.org)

File: db/upgrade-X.X.X-X.Y.Z.sql [hunk 3]
```diff
@@ -316,3 +316,15 @@ DROP INDEX mac ON violation;
 --
 
 DROP INDEX ip on inline_accounting;
+
+--
+-- Alter table iplog_history
+--
+
+ALTER TABLE iplog_history ADD INDEX end_time (end_time) ADD INDEX start_time (start_time);
```
Comment (dwlfrth) [inline db/upgrade-X.X.X-X.Y.Z.sql]: ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'ADD INDEX start_time (start_time)' at line 1

Missing comma before second ADD

File: db/upgrade-X.X.X-X.Y.Z.sql [hunk 4]
```diff
@@ -316,3 +316,15 @@ DROP INDEX mac ON violation;
...
+
+--
+-- Alter table iplog_archive
+--
+
+ALTER TABLE iplog_archive ADD INDEX end_time (end_time) ADD INDEX start_time (start_time);
```
Comment (dwlfrth) [inline db/upgrade-X.X.X-X.Y.Z.sql]: ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'ADD INDEX start_time (start_time)' at line 1

Missing comma before second ADD

**Review by dwlfrth — CHANGES_REQUESTED** [review]

**Comment by jrouzierinverse** [comment]
Fixed

James Rouzier
jrouzier@inverse.ca :: +1.514.447.4918 (x115)  ::  http://www.inverse.ca
Inverse inc. :: Leaders behind SOGo (http://www.sogo.nu) and PacketFence (http://www.packetfence.org)

**Pull request merged**